### PR TITLE
Token cleanup

### PR DESCRIPTION
**Pull request by blen-desta**
- Deleted the unused BSTAuthenticationToken
- Deleted the unnecessary MockAuthenticationToken
- Made the BaseAuthenticationToken abstract
- Changed BasicAuthenticationHandler to use an STSAuthenticationToken

### CODE REVIEW

File: platform/security/sts/security-sts-bstdelegationhandler/src/main/java/org/codice/ddf/security/delegation/BSTDelegationHandler.java [hunk 1]
```diff
@@ -30,19 +28,13 @@
  */
 public class BSTDelegationHandler implements TokenDelegationHandler {
```
Comment (SmithJosh) [inline platform/security/sts/security-sts-bstdelegationhandler/src/main/java/org/codice/ddf/security/delegation/BSTDelegationHandler.java:29]: ❓ Is this still needed?

Comment (blen-desta) [reply platform/security/sts/security-sts-bstdelegationhandler/src/main/java/org/codice/ddf/security/delegation/BSTDelegationHandler.java:29]: I'm not sure that's why I just left it

**Review by SmithJosh — APPROVED** [review]

File: platform/security/sts/security-sts-bstdelegationhandler/pom.xml [hunk 2]
```diff
@@ -71,6 +71,43 @@
...
+                        <configuration>
+                            <haltOnFailure>true</haltOnFailure>
+                            <rules>
+                                <rule>
+                                    <element>BUNDLE</element>
+                                    <limits>
+                                        <limit implementation="org.codice.jacoco.LenientLimit">
+                                            <counter>INSTRUCTION</counter>
+                                            <value>COVEREDRATIO</value>
+                                            <minimum>0.99</minimum>
```
Comment (bakejeyner) [inline platform/security/sts/security-sts-bstdelegationhandler/pom.xml:92]: 🚀

File: platform/security/sts/security-sts-bstdelegationhandler/src/main/java/org/codice/ddf/security/delegation/BSTDelegationHandler.java (hunk 1 above)
Comment (bakejeyner) [reply platform/security/sts/security-sts-bstdelegationhandler/src/main/java/org/codice/ddf/security/delegation/BSTDelegationHandler.java:29]: Wo. Looks like the two implementing classes we have for the `TokenDelegationHandler` interface are `BSTDelegationHandler` and `x509DelegationHandler`, which both do the same thing.

They both allow delegation if the given token parameters have a `BinarySecurityToken` on them. What that _actually_ means; I have no idea 🤷‍♂. But it sounds like this class might be able to be combined with the other class / deleted.

**Review by bakejeyner — APPROVED** [review]

File: platform/security/sts/security-sts-bstdelegationhandler/src/main/java/org/codice/ddf/security/delegation/BSTDelegationHandler.java (hunk 1 above)
Comment (bakejeyner) [reply platform/security/sts/security-sts-bstdelegationhandler/src/main/java/org/codice/ddf/security/delegation/BSTDelegationHandler.java:29]: Also, most of the code in `X509DelegationHandler` is commented out 😆

**Review by stustison — APPROVED** [review]

**Pull request merged**